### PR TITLE
[fast-ut] [reduced-it] Add ArrayAggregate and ArraySort corner cases [databricks]

### LINKED ISSUE #15616: [AutoSparkUT] Add ArrayAggregate and ArraySort corner-case IT coverage
## Coverage gap

The existing ArraySort and ArrayAggregate Python integration tests exercise their basic supported paths, but they did not explicitly pin these edge values:

- ArraySort: finite Float/Double extrema, positive and negative zero, NaN, empty/blank strings, ASCII, CJK, and emoji strings.
- ArrayAggregate: Byte/Short boundaries plus Float/Double SUM and PRODUCT with finite extrema, positive and negative zero, and NaN.

Positive and negative infinity are intentionally out of scope for this corner-case milestone.

## Test change

Extend the existing parameterized tests rather than add duplicate test functions. The ArrayAggregate cases cover every currently GPU-supported aggregate state/input family (Boolean and the integral/floating-point types); unsupported Decimal, Date, Timestamp, and String paths are not counted as GPU coverage. The Float/Double cases enable `spark.rapids.sql.variableFloatAgg.enabled=true` and use one-element non-empty arrays so strict CPU/GPU equality is deterministic while still exercising every listed corner value. Every ArrayAggregate parameter now requires `GpuArrayAggregate`, and ArraySort continues to require `GpuArraySort`.

## Local validation evidence

Spark 3.3.0 / Python 3.10, current branch build:

```text
collection_ops_test.py::test_array_sort
higher_order_functions_test.py::test_array_aggregate_numeric_ops
51 passed, 24 warnings in 58.28s
```

Spark 3.5.0 / shim350, JaCoCo run against nightly anchor `4f725f15`:

```text
51 passed, 39703 deselected, 12 warnings in 36.89s
covered lines: +0
```

### PR DESCRIPTION
**Pull request by wjxiz1992**
**JaCoCo sql-plugin line coverage: +0 lines** (shim350 vs nightly b14; ArrayAggregate and ArraySort production paths were already covered)

Closes #15616

## Summary

Extend the existing Python integration tests for the two corner-case targets:

- `ArraySort`: Float and Double finite extrema, positive and negative zero, NaN, plus empty, blank, ASCII, CJK, and emoji strings. The capture assertion requires `GpuArraySort`.
- `ArrayAggregate`: explicit Byte/Short boundaries and Float/Double SUM/PRODUCT over finite extrema, positive and negative zero, and NaN. Together with the existing cases, every currently GPU-supported aggregate state/input family is covered; unsupported Decimal, Date, Timestamp, and String paths are not counted. Every parameter requires `GpuArrayAggregate`.

Positive and negative infinity are intentionally out of scope for this milestone. Existing test functions were extended inline so the coverage scanner can attribute the literal generators without duplicating test scaffolding. Float/Double aggregation enables `spark.rapids.sql.variableFloatAgg.enabled=true` and uses one-element non-empty arrays for deterministic strict CPU/GPU comparison.

## Validation

Spark 3.3.0 / Python 3.10, current branch build:

```text
collection_ops_test.py::test_array_sort
higher_order_functions_test.py::test_array_aggregate_numeric_ops
51 passed, 24 warnings in 58.28s
```

Spark 3.5.0 / shim350, JaCoCo run against nightly anchor `4f725f15`:

```text
51 passed, 39703 deselected, 12 warnings in 36.89s
covered lines: +0
```

Documentation
- [ ] Updated for new or modified user-facing features or behaviors
- [x] No user-facing change

Testing
- [x] Added or modified tests to cover new code paths
- [ ] Covered by existing tests
      (Please provide the names of the existing tests in the PR description.)
- [ ] Not required

Performance
- [ ] Tests ran and results are added in the PR description
- [ ] Issue filed with a link in the PR description
- [x] Not required

### CODE REVIEW

File: integration_tests/src/main/python/higher_order_functions_test.py [hunk 1]
```diff
@@ -35,19 +35,41 @@ def do_project(spark):
...
+        '(acc, x) -> acc + CAST(x as BIGINT)', '0L'),
+    (ArrayGen(IntegerGen(min_val=-3, max_val=3), max_length=8),
+        '(acc, x) -> acc * CAST(x as BIGINT)', '1L'),
+    (ArrayGen(IntegerGen(min_val=-100, max_val=100), max_length=8),
+        '(acc, x) -> greatest(acc, CAST(x as BIGINT))', '-9223372036854775808L'),
+    (ArrayGen(IntegerGen(min_val=-100, max_val=100), max_length=8),
+        '(acc, x) -> least(acc, CAST(x as BIGINT))', '9223372036854775807L'),
+    (ArrayGen(FloatGen(
+        no_nans=True,
+        special_cases=[FLOAT_MIN, FLOAT_MAX, 0.0, -0.0, float('nan')]), max_length=1),
```
[the remark this answers is not in the document]
Comment (wjxiz1992) [reply integration_tests/src/main/python/higher_order_functions_test.py]: Thanks, updated.

**Comment by wjxiz1992** [comment]
build

File: integration_tests/src/main/python/higher_order_functions_test.py [hunk 2]
```diff
@@ -35,19 +36,54 @@ def do_project(spark):
...
+        '(acc, x) -> least(acc, CAST(x as BIGINT))', '9223372036854775807L'),
+    (ArrayGen(
+        ByteGen(special_cases=[BYTE_MIN, BYTE_MAX, 0, 1, -1]),
+        min_length=1, max_length=8),
+        '(acc, x) -> acc + x', 'CAST(0 AS TINYINT)'),
+    (ArrayGen(
+        ShortGen(special_cases=[SHORT_MIN, SHORT_MAX, 0, 1, -1]),
+        min_length=1, max_length=8),
+        '(acc, x) -> acc + x', 'CAST(0 AS SMALLINT)'),
+    (ArrayGen(FloatGen(
```
Comment (thirtiseven) [inline integration_tests/src/main/python/higher_order_functions_test.py:58]: These cases include `-0.0`, but the shared float oracle ultimately uses `lhs == rhs`, which treats `+0.0` and `-0.0` as equal. As a result, ArraySort could change their relative representation, or PRODUCT could lose the negative-zero sign, without failing. Could we make the signed-zero cases deterministic and compare their sign/raw bits while retaining the GPU plan assertion?

**Comment by wjxiz1992** [comment]
build

**Comment by wjxiz1992** [comment]
build

File: integration_tests/src/main/python/higher_order_functions_test.py (hunk 2 above)
Comment (wjxiz1992) [reply integration_tests/src/main/python/higher_order_functions_test.py:58]: Updated.

**Comment by wjxiz1992** [comment]
build

**Comment by wjxiz1992** [comment]
@firestarman Could you please review this PR when you have a chance? Thanks.

File: integration_tests/src/main/python/collection_ops_test.py [hunk 3]
```diff
@@ -167,7 +167,15 @@ def test_sort_array_lit(data_gen):
 
 # array_sort(arr) with the default comparator (ascending, nulls last) must run on GPU.
 # Capture the plan so a silent CPU fallback (which still matches results) fails the test.
-@pytest.mark.parametrize('data_gen', _array_sort_gens, ids=idfn)
+@pytest.mark.parametrize('data_gen', _array_sort_gens + [
```
Comment (thirtiseven) [inline integration_tests/src/main/python/collection_ops_test.py]: Why not edit the `_array_sort_gens` itself?

**Review by thirtiseven — DISMISSED** [review]
a non-blocking nit/question.

**Comment by wjxiz1992** [comment]
build

**Review by thirtiseven — APPROVED** [review]
Thanks.
